src/Broadway/Domain/Metadata.php

```diff
@@ -43,24 +56,37 @@ public function merge(Metadata $otherMetadata)
     }
 
     /**
-     * {@inheritDoc}
+     * Returns an array with all metadata.
+     *
+     * @return array
      */
-    public function serialize()
+    public function all()
     {
         return $this->values;
     }
 
     /**
-     * Helper method to construct an instance containing the key and value.
+     * Get a specific metadata value based on key.
      *
-     * @param mixed $key
-     * @param mixed $value
+     * @param string $key
      *
-     * @return Metadata
+     * @return mixed
      */
-    public static function kv($key, $value)
+    public function get($key)
```

Comment:
instead of returning null should we return a specified default value like in ParameterBag? https://github.com/symfony/http-foundation/blob/v3.3.6/ParameterBag.php#L86-L89

It's more functional that way IMO.